### ISSUE TITLE
managed-groups don't seem to actually get role grants

### ISSUE THREAD

**Issue by randriksen** [hashicorp/boundary#1771]
Not sure if anyone else has had this problem, but i haven't found the exact same one in here.

**Describe the bug**
It seems like when trying to add managed-groups as principals in roles, it doesn't actually grant the users in said groups any rights.
My guess is that it's because the managed-groups use the account objects and not the user objects.

**To Reproduce**
1. setup boundary with an oidc
2. set the oidc to map users with managed-groups
3. add managed-groups as principals in roles
4. login with oidc and confirm that the user is actually in said managed group (probably with an admin user)
5. not being able to do anyting with the oidc-user

**Expected behavior**
i would expect the user to have the grants from the role it's supposed to have and be able to do the things that it's allowed

**Additional context**
if I explicitly add the user object or a normal group as a principle in the role, it works as expected. it's only a problem with managed groups

**Comment by micchickenburger** [comment]
The only way I got this to work was to create managed groups and roles in Terraform:
```hcl
# Add access control groups
resource "azuread_group" "app_users" {
  display_name            = "app-users"
  description             = "App Users (Managed by Terraform)"
  security_enabled        = true # security group, as opposed to an Office 365 group
  prevent_duplicate_names = true
}

resource "boundary_managed_group" "app_users" {
  name           = "app-user-group"
  description    = "App Users"
  auth_method_id = var.azure_auth_id
  filter         = "\"${azuread_group.app_users.id}\" in \"/token/groups\""
}

# Add role for accessing app
resource "boundary_role" "app_user_access" {
  name           = "app-user-role"
  description    = "Access to App"
  scope_id       = var.scope_id
  grant_scope_id = boundary_scope.app.id
  grant_strings = [
    "id=*;type=session;actions=read:self,cancel:self",
    "type=target;actions=list",
    "id=${boundary_target.app_web.id};actions=read,authorize-session",
  ]
  principal_ids = [boundary_managed_group.app_users.id] # set here, but won't show up in UI under Role Principals
}
```

**Comment by jefferai** [comment]
> The grants seem to apply to managed groups at the org scope but not the project scope.

Roles only apply grants to a single scope, which defaults to the same scope containing the role. `grant_scope_id` can be used to explicitly specify the same scope or a downstream scope.

**Comment by randriksen** [comment]
That's interesting, but I'm still having problems getting the right roles tied to the right users. I'm still seeing that the roles need to be tied to users, while the managed groups contain accounts, not users.

i'm trying to set this up with terraform, and here's an anonymized version of what i'm working with. probably won't work, but if you have any ideas what i should change or add to it, i'd love to try.

`

resource "boundary_managed_group" "all_users_group" {
  name           = "All  users"
  description    = "catches all users with @cow.com in their email"
  auth_method_id = boundary_auth_method_oidc.provider.id
  filter         = "email == \"@cow.com\""
}


resource "boundary_managed_group" "cow" {
  name           = "COW"
  description    = "Mooo!"
  auth_method_id = boundary_auth_method_oidc.provider.id
  filter         = "roles contains \"COW\""
}
variable "login_server_ips" {
  type    = set(string)
  default = [
    "login-osl1.cow.net",
    "login-osl2.cow.net",
  ]
}

variable "rds01_w_cow_net_ips" {
  type    = set(string)
  default = [
      "rds01.cow.net"
  ]
}

resource "boundary_scope" "global" {
  global_scope = true
  description  = "My first global scope!"
  scope_id     = "global"
}

resource "boundary_scope" "cow" {
  name                     = "cow CloudOps"
  description              = "Core scope"
  scope_id                 = boundary_scope.global.id
  auto_create_admin_role   = true
  auto_create_default_role = true
}


resource "boundary_role" "global_admin" {
  name          = "global admin"
  description   = "global admin"
  principal_ids = [boundary_managed_group.all_cow_group.id]
  grant_strings = ["id=*;type=*;actions=*"]
  scope_id      = boundary_scope.global.id
}

resource "boundary_role" "global_admin_app_access" {
  name           = "global_admin_appaccess"
  scope_id       = boundary_scope.global.id
  grant_scope_id = boundary_scope.cow_cow.id
  grant_strings = ["id=*;type=*;actions=*"]
  principal_ids  = [boundary_managed_group.cow.id, boundary_managed_group.all_cow_group.id]

}

resource "boundary_role" "cow_admin_app_access" {
  name           = "cow_admin_appaccess"
  scope_id       = boundary_scope.cow.id
  grant_scope_id = boundary_scope.cow_cow.id
  grant_strings = ["id=*;type=*;actions=*"]
  principal_ids  = [boundary_managed_group.cow.id, boundary_managed_group.all_cow_group.id]

}
resource "boundary_role" "cow_admin" {
  name          = "cow admin"
  description   = "cow admin"
  principal_ids = [boundary_managed_group.all_cow_group.id]
  grant_strings = ["id=*;type=*;actions=*"]
  scope_id      = boundary_scope.cow.id
}

resource "boundary_role" "cow_bastions_admin" {
  name          = "cow_bastions admin"
  description   = "cow bastions admin"
  principal_ids = [boundary_managed_group.all_cow_group.id]
  grant_strings = ["id=*;type=*;actions=*"]
  scope_id      = boundary_scope.cow_bastions.id
}

resource "boundary_role" "cow_admin" {
  name          = "cow admin"
  description   = "cow admin"
  principal_ids = [boundary_managed_group.cow.id, boundary_managed_group.all_cow_group.id]
  grant_strings = ["id=*;type=*;actions=*"]
  scope_id      = boundary_scope.cow_cow.id
}

resource "boundary_scope" "cow_bastions" {
  name                   = "cow Bastion hosts"
  description            = "This is where we should find bastion hosts"
  scope_id               = boundary_scope.cow.id
  auto_create_admin_role = true
}


resource "boundary_scope" "cow_cow" {
  name                   = "COW"
  description            = "Mostly windows servers"
  scope_id               = boundary_scope.cow.id
  auto_create_admin_role = true
  
}

resource "boundary_host_catalog" "login_servers" {
  name        = "login_servers"
  description = "login servers host catalog"
  type        = "static"
  scope_id    = boundary_scope.cow_bastions.id
}

resource "boundary_host_catalog" "w_cow_net" {
    name = "w_cow_net"
    description = "cow.net host catalog"
    type = "static"
    scope_id = boundary_scope.cow_bastions.id
}

resource "boundary_host" "rds01_w_cow_net" {
    for_each = var.rds01_w_cow_net_ips
    type = "static"
    name = "rds01_w_cow_net"
    description = "rds01.cow.net"
    address = each.key
    host_catalog_id = boundary_host_catalog.w_cow_net.id
}

resource "boundary_host" "login_servers" {
  for_each        = var.login_server_ips
  type            = "static"
  name            = "login_server_service_${each.value}"
  description     = "login server host"
  address         = each.key
  host_catalog_id = boundary_host_catalog.login_servers.id
}

resource "boundary_host_set" "login_servers_ssh" {
  type            = "static"
  name            = "login_servers_ssh"
  description     = "Host set for login servers"
  host_catalog_id = boundary_host_catalog.login_servers.id
  host_ids        = [for host in boundary_host.login_servers : host.id]
}

resource "boundary_host_set" "rds01_w_cow_net_rdp" {
    type = "static"
    name = "rds01_w_cow_net_rdp"
    description = "rds01.cow.net"
    host_catalog_id = boundary_host_catalog.w_cow_net.id
    host_ids = [for host in boundary_host.rds01_w_cow_net : host.id]
}


resource "boundary_target" "rds01_w_cow_net_rdp" {
    type = "tcp"
    name = "rds01_w_cow_net_rdp"
    description = "rds01 target"
    scope_id = boundary_scope.cow_bastions.id
    default_port = "3389"

    host_set_ids = [
         boundary_host_set.rds01_w_cow_net_rdp .id
    ]
}

# create target for accessing login servers on port :22
resource "boundary_target" "login_servers_ssh" {
  type         = "tcp"
  name         = "ssh_server"
  description  = "login SSH target"
  scope_id     = boundary_scope.cow_bastions.id
  default_port = "22"

  host_set_ids = [
    boundary_host_set.login_servers_ssh.id
  ]
}
`

**Comment by jefferai** [comment]
It's honestly hard to follow the terraform, but here are some CLI steps I captured. It starts with logging in as the unprivileged user against a dev instance and being unable to list any targets. I then switch to the admin user, add a managed group, add a role, add grants and the managed group to the role, and then log in again with the unprivileged user and am able to list targets in the project scope.

Your problem is almost certainly the filter. `email` is not a valid selector. Selectors use JSON Pointer which means they must always start with a / and be quoted. See https://www.boundaryproject.io/docs/concepts/filtering/oidc-managed-groups for some examples here. Additionally, as per the documentation you need to be specifying token or userinfo claims as the first part of the selector.

```
$ boundary targets list -scope-id p_1234567890
Error from controller when performing list on targets

Error information:
  Kind:                PermissionDenied
  Message:             Forbidden.
  Status:              403
  context:             Error from controller when performing list on targets


$ boundary authenticate oidc -auth-method-id amoidc_1234567890
Opening returned authentication URL in your browser...

Authentication information:
  Account ID:      acctoidc_1234567890
  Auth Method ID:  amoidc_1234567890
  Expiration Time: Thu, 13 Jan 2022 13:25:16 EST
  User ID:         u_1234567890

The token was successfully stored in the chosen keyring and is not displayed here.

$ boundary managed-groups create oidc -auth-method-id amoidc_1234567890 -filter '"/token/sub" == "user"'

Managed Group information:
  Auth Method ID:      amoidc_1234567890
  Created Time:        Thu, 06 Jan 2022 13:26:31 EST
  ID:                  mgoidc_N0AmVt6ZBQ
  Type:                oidc
  Updated Time:        Thu, 06 Jan 2022 13:26:31 EST
  Version:             1

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete

  Attributes:
    Filter:            "/token/sub" == "user"

$ boundary roles create -scope-id p_1234567890

Role information:
  Created Time:        Thu, 06 Jan 2022 13:27:20 EST
  Grant Scope ID:      p_1234567890
  ID:                  r_rZXKfTPOt3
  Updated Time:        Thu, 06 Jan 2022 13:27:20 EST
  Version:             1

  Scope:
    ID:                p_1234567890
    Name:              Generated project scope
    Parent Scope ID:   o_1234567890
    Type:              project

  Authorized Actions:
    no-op
    read
    update
    delete
    add-principals
    set-principals
    remove-principals
    add-grants
    set-grants
    remove-grants

$ boundary roles add-grants -id r_rZXKfTPOt3 -grant "id=*;type=target;actions=list,read"

Role information:
  Created Time:        Thu, 06 Jan 2022 13:27:20 EST
  Grant Scope ID:      p_1234567890
  ID:                  r_rZXKfTPOt3
  Updated Time:        Thu, 06 Jan 2022 13:28:23 EST
  Version:             2

  Scope:
    ID:                p_1234567890
    Name:              Generated project scope
    Parent Scope ID:   o_1234567890
    Type:              project

  Authorized Actions:
    no-op
    read
    update
    delete
    add-principals
    set-principals
    remove-principals
    add-grants
    set-grants
    remove-grants

  Canonical Grants:
    id=*;type=target;actions=list,read

$ boundary roles add-principals -id r_rZXKfTPOt3 -principal mgoidc_N0AmVt6ZBQ

Role information:
  Created Time:        Thu, 06 Jan 2022 13:27:20 EST
  Grant Scope ID:      p_1234567890
  ID:                  r_rZXKfTPOt3
  Updated Time:        Thu, 06 Jan 2022 13:28:55 EST
  Version:             3

  Scope:
    ID:                p_1234567890
    Name:              Generated project scope
    Parent Scope ID:   o_1234567890
    Type:              project

  Authorized Actions:
    no-op
    read
    update
    delete
    add-principals
    set-principals
    remove-principals
    add-grants
    set-grants
    remove-grants

  Principals:
    ID:             mgoidc_N0AmVt6ZBQ
      Type:         managed group
      Scope ID:     global

  Canonical Grants:
    id=*;type=target;actions=list,read

$ boundary authenticate oidc -auth-method-id amoidc_1234567890
Opening returned authentication URL in your browser...

Authentication information:
  Account ID:      acctoidc_0987654321
  Auth Method ID:  amoidc_1234567890
  Expiration Time: Thu, 13 Jan 2022 13:29:31 EST
  User ID:         u_0987654321

The token was successfully stored in the chosen keyring and is not displayed here.

$ boundary targets list -scope-id p_1234567890

Target information:
  ID:                    ttcp_1234567890
    Version:             2
    Type:                tcp
    Name:                Generated target
    Description:         Provides an initial target in Boundary
    Authorized Actions:
      read
      authorize-session
```

**Comment by randriksen** [comment]
Yeah, the terraform isn't easily read, sorry about that.
The filter i used actually seems to work for me, this is part of the data
that is sent with the oidc data and when i tried to add anything else into
the filter, it didn't work. We're using keycloak for oidc.
Users are able to log in and i can see them in the managed group in cli.
What doesn't work is that they don't get the grants they should have.

I'll try to follow the cli you suggested and see what happens.

**Comment by jwueste** [comment]
Did you find a solution or a workaround for this issue?
I've got the same problem, which is currently blocking my POC... :(

**Comment by jefferai** [comment]
@jwueste The filters that @randriksen were using will never work, so at the moment this isn't a valid issue -- which is to say that I can't really help with your case without more information. I suggest opening a new issue and I can help you out with your specific configuration.

**Comment by randriksen** [comment]
> @jwueste The filters that @randriksen were using will never work, so at the moment this isn't a valid issue -- which is to say that I can't really help with your case without more information. I suggest opening a new issue and I can help you out with your specific configuration.

I'm sorry to say that the filters I was using actually worked fine,
<img width="550" alt="filtertest" src="https://user-images.githubusercontent.com/1292372/152517501-2c1e3a48-61ef-4c70-9640-333a8cd3f170.png">
<img width="811" alt="filtertest2" src="https://user-images.githubusercontent.com/1292372/152517503-ae426396-7150-49d9-bba4-58a48b42c064.png">

The filters for the group memberships were **never** the problem. And as far as i can see, the problem isn't just with managed groups, but also with users authenticating with passwords and where groups aren't even in the picture.

As far as I have found so far the problem is related to rights and how they don't propagate properly.
The global admin doesn't automatically inherit rights to subscopes. the only user that gets automatic rights is the one that creates the resource in the first place. everything else i've had to specifically grant.




```
  #create a new scope
  boundary scopes create -name test -scope-id global

Scope information:
  Created Time:        Fri, 04 Feb 2022 11:02:03 UTC
  ID:                  o_vc6qGDM2Rz
  Name:                test
  Updated Time:        Fri, 04 Feb 2022 11:02:03 UTC
  Version:             1

  Scope (parent):
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete

#create a new role for scope
  boundary roles create -name "test admin" -scope-id global -grant-scope-id o_vc6qGDM2Rz

Role information:
  Created Time:        Fri, 04 Feb 2022 11:02:33 UTC
  Grant Scope ID:      o_vc6qGDM2Rz
  ID:                  r_K4d7V8mIDz
  Name:                test admin
  Updated Time:        Fri, 04 Feb 2022 11:02:33 UTC
  Version:             1

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete
    add-principals
    set-principals
    remove-principals
    add-grants
    set-grants
    remove-grants

#make the new role admin of the new scope
  boundary roles add-grants -id r_K4d7V8mIDz -grant "id=*;type=*;actions=*"

Role information:
  Created Time:        Fri, 04 Feb 2022 11:02:33 UTC
  Grant Scope ID:      o_vc6qGDM2Rz
  ID:                  r_K4d7V8mIDz
  Name:                test admin
  Updated Time:        Fri, 04 Feb 2022 11:02:54 UTC
  Version:             2

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete
    add-principals
    set-principals
    remove-principals
    add-grants
    set-grants
    remove-grants

  Canonical Grants:
    id=*;type=*;actions=*

#create a new user in global
  boundary users create -scope-id global -name test

User information:
  Created Time:        Fri, 04 Feb 2022 11:04:21 UTC
  ID:                  u_jH5kXW77Xm
  Name:                test
  Updated Time:        Fri, 04 Feb 2022 11:04:21 UTC
  Version:             1

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete
    add-accounts
    set-accounts
    remove-accounts


#create a new account with password as authentication
  boundary accounts create password -auth-method-id ampw_KvJWMZouKO -name test -login-name test
Password is not set as flag, please enter it now (will be hidden):
Please enter it one more time for confirmation:


Account information:
  Auth Method ID:      ampw_KvJWMZouKO
  Created Time:        Fri, 04 Feb 2022 11:05:27 UTC
  ID:                  acctpw_zqR2YAXCHx
  Name:                test
  Type:                password
  Updated Time:        Fri, 04 Feb 2022 11:05:27 UTC
  Version:             1

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete
    set-password
    change-password

  Attributes:
    Login Name:        test


 #attach account to the new user
  boundary users add-accounts -id u_jH5kXW77Xm -account acctpw_zqR2YAXCHx

User information:
  Created Time:        Fri, 04 Feb 2022 11:04:21 UTC
  ID:                  u_jH5kXW77Xm
  Name:                test
  Updated Time:        Fri, 04 Feb 2022 11:06:10 UTC
  Version:             2

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete
    add-accounts
    set-accounts
    remove-accounts

  Accounts:
    ID:                acctpw_zqR2YAXCHx
    Scope ID:          global

#attach new user to original administration role for global
#Administration
#Provides admin grants within the "global" scope to the initial user
#
#r_Ca2jvEIHq9 

  boundary roles add-principals -id r_Ca2jvEIHq9 -principal u_jH5kXW77Xm

Role information:
  Created Time:        Mon, 24 Jan 2022 11:56:22 UTC
  Description:         Provides admin grants within the "global" scope to the initial user
  Grant Scope ID:      global
  ID:                  r_Ca2jvEIHq9
  Name:                Administration
  Updated Time:        Fri, 04 Feb 2022 11:06:59 UTC
  Version:             12

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete
    add-principals
    set-principals
    remove-principals
    add-grants
    set-grants
    remove-grants

  Principals:
    ID:             u_jH5kXW77Xm
      Type:         user
      Scope ID:     global
    ID:             u_29ADaKPsyD
      Type:         user
      Scope ID:     global

  Canonical Grants:
    id=*;type=*;actions=*
 

```
 Try to log in with this user and see if you have admin of the test scope... you shouldn't

```
# now attach the new role to the user
boundary roles add-principals -id r_K4d7V8mIDz -principal u_jH5kXW77Xm

Role information:
  Created Time:        Fri, 04 Feb 2022 11:02:33 UTC
  Grant Scope ID:      o_vc6qGDM2Rz
  ID:                  r_K4d7V8mIDz
  Name:                test admin
  Updated Time:        Fri, 04 Feb 2022 11:13:51 UTC
  Version:             3

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    no-op
    read
    update
    delete
    add-principals
    set-principals
    remove-principals
    add-grants
    set-grants
    remove-grants

  Principals:
    ID:             u_jH5kXW77Xm
      Type:         user
      Scope ID:     global

  Canonical Grants:
    id=*;type=*;actions=*
```

    now you should be able to make new projects in the test scope. 
but if you create projects with the test user, the admin user won't be able to see them, and vice versa.

it does seem that the problem is worse with managed-groups than with just users, since if i add managed groups to the admin roles, they just don't seem to get them at all.
even if i set the filter to` "\"/userinfo/email\" == \"@cow.com\""` which just isn't neccessary.

**Comment by aadeshrao** [comment]
Hello, I was facing a similar issue, but then when I looked closely into @jefferai steps, he had created the role inside the project scope. `boundary roles create -scope-id p_1234567890`. Whereas @randriksen we have been creating the role with our org scope `boundary roles create -name "test admin" -scope-id global -grant-scope-id o_vc6qGDM2Rz`. I am not sure if you had intended for the role to be in the org scope vs project scope.

The reason why users won't probably see targets or hosts is because their scope is projects and not orgs. Refer - https://www.boundaryproject.io/docs/concepts/security/permissions#resource-table
Applicable scopes for Roles, and resources under host-catalogs is project and not the org. Even though projects are inside orgs, projects hold precedence. 

Not sure if this helps, or if am way off base!

**Comment by jefferai** [comment]
>As far as I have found so far the problem is related to rights and how they don't propagate properly.

Scopes are permission boundaries. Permissions don't propagate into child scopes -- they have to be granted. This has a number of benefits, but the main is that it removes the need for `deny`. Long time experience with Vault users showed that `deny` makes it very complicated to understand the final set of permissions. In Boundary they are only additively composable, which significantly reduces the chance of granting permissions accidentally. By making scopes permission boundaries you don't need to also include `deny` to stop propagation of permissions.

If you don't skip creating the admin role when a scope is created then a role will automatically be set up to give the creating user administrative privileges in the scope, but outside of that (and the default role in the scope, also if not skipped) only the actual granted permissions in a scope apply.

Note that you can use `grant_scope_id` to create a role in a parent scope that affects a child scope.

**Comment by Rosslu** [comment]
did anyone figure this out?

If I create a user and add to a normal group and add the group id as a principle the users in that group gets access, if I add the managed group id as principles I can verify that the user is in the managed group and that the managed group id is listed as a principle but the user doesn't get any permissions relating to the role. also when I run  boundary roles read -id r_zwQlA5NukJ the principles will show up for managed group but not in the ui

**Comment by andel7** [comment]
Have the same issue as @Rosslu.  I see oidc managed group with members in it, and its configured as  principal in boundary role. When oidc user logged in he gets no permissions. If I add this oidc user explicitly as principal in the boundary role I get the expected permissions.

**Comment by jefferai** [comment]
@Rosslu @andel7 Could one of you file a new bug? This issue was not a bug in Boundary but a configuration issue. Rather than revive it for a different problem please open a new issue. Thanks!

**Comment by Rosslu** [comment]
I managed a workaround but maybe at least the documentation needs to be updated, here is my terraform
`resource "boundary_managed_group" "staging" {
  name           = "default staging group"
  description    = "An example of an OIDC managed group"
  auth_method_id = boundary_auth_method_oidc.staging.id
  filter         = "\"DevOps\" in \"/userinfo/groups\""
}`

although the object gets returned as 
`{
  "sub": "1f09ddc2-5ada-4829-1234-368a446d6f17",
  "email_verified": true,
  "name": "Joe Soap",
  "groups": [
    "DevOps"
  ],
  "preferred_username": "joe.soap@soapiejoes.com",
  "given_name": "Joe",
  "family_name": "Soap",
  "email": "joe.soap@soapiejoes.com"
}`

- I noticed that the username will show up in the managed group but when I query it from the boundary cli it didn't show up. 
- The reason it didn't show up is because the filter looks in the wrong place 
- when I listed the user object from the cli I noticed it's embedded in the userinfo/groups not just groups and that fixed the issue.

**Comment by andel7** [comment]
Despite members showing up in OIDC managed group my issue was filters
Here's a Hashicorp discuss topic I opened with all the details and Jeff provided the solution:
https://discuss.hashicorp.com/t/oidc-managed-groups-dont-get-proper-permissions/43933

**Comment by jefferai** [comment]
@Rosslu Can you explain a bit more? What do you mean by managed a workaround? That filter and those OIDC claims look to be a perfect match 👌 

I'm also not sure what you mean by the rest of it:

>I noticed that the username will show up in the managed group but when I query it from the boundary cli it didn't show up.

Managed groups don't get usernames attached to them, they get accounts attached to them.

>The reason it didn't show up is because the filter looks in the wrong place

Which filter? Which wrong place? What you have there seems perfectly good.

>when I listed the user object from the cli I noticed it's embedded in the userinfo/groups not just groups and that fixed the issue.

Do you mean the account object? User objects don't show JWT information as they are tied to one or more accounts, such as an OIDC account.

**Comment by Rosslu** [comment]
```
 resource "boundary_managed_group" "org" {
  name           = "DevOps global group"
  description    = "An example of an OIDC managed group"
  auth_method_id = boundary_auth_method_oidc.oidcs.id
  filter         = "\"DevOps\" in \"/userinfo/groups\""
} 
```
Notice the escape in the filter. 

Also the object returned from my oidc is
```
{
  "sub": "1f09ddc2-1234-56789-b6f8-368a446d6f17",
  "email_verified": true,
  "name": "Joe Soap",
  "groups": [
    "DevOps"
  ],
  "preferred_username": "joesoap@soapiejoe.com",
  "given_name": "Joe",
  "family_name": "Soap",
  "email": "joesoap@soapiejoe.com"
}
```
this object gets added to the userinfo object inside boundary so to filter for groups DevOps. 

I figured it out by using 
`boundary accounts read -id acctoidc_mMFhMNtSOA <The value of the members in managed group>`
*you'll notice it says  userinfo_claims but when you filter you use just userinfo. 
This and the escape values sorted me out. 

Then add the managed group id in your case mgoidc_N0AmVt6ZBQ as a principle id to your grant. 
You can view this with the boundary cli but they don't show up when you look in the ui.

Hope it makes sense

**Comment by jefferai** [comment]
Hi @Rosslu ,

Correct, when looking at an account it shows you `userinfo_claims` and `token_claims` to make it clear both what they are (claims) and where they came from (token or userinfo). However, [as per the docs](https://www.boundaryproject.io/docs/concepts/filtering/oidc-managed-groups#oidc-managed-groups-filtering), those are put into `/token` and `/userinfo`.

Glad this is working for you!

**Comment by Tristan971** [comment]
I'm somewhat confused about this issue being closed... It is very much still a problem, at least one of documentation.

Spent the better part of a day wrestling with this useless account:
```
$ boundary accounts read -id acctoidc_svodlq5Qm7

Account information:
  Auth Method ID:      amoidc_oo6rTM3tGF
  Created Time:        Sun, 27 Nov 2022 19:39:11 GMT
  ID:                  acctoidc_svodlq5Qm7
  Type:                oidc
  Updated Time:        Sun, 27 Nov 2022 23:55:06 GMT
  Version:             7

  Scope:
    ID:                global
    Name:              global
    Type:              global

  Authorized Actions:
    read

  Attributes:
    ...
    token_claims:
    {
    ...
    "sub": "63694225-a481-43ba-b6a6-f60ff45d143d",
    ...
    }
    userinfo_claims:
    {
    ...
    "sub": "63694225-a481-43ba-b6a6-f60ff45d143d",
    ...
    }
```

For which the following OIDC group filters (not expliciting quote-escaping for lisibility):
- matches with `"/token/sub" == "63694225-a481-43ba-b6a6-f60ff45d143d"`
- doesn't match with `"/userinfo/sub" == "63694225-a481-43ba-b6a6-f60ff45d143d"`

And I still have absolutely no idea why, to be honest. It is also not specific to the "sub" claim, other claims also behave the same way.

My best guess is that it somehow relates to some side effect of the claims_scopes from this documentation example:
![2022-11-28 00_51_47](https://user-images.githubusercontent.com/875533/204169964-14eb9ef3-4c16-4c7c-a8dc-8f685ce3261a.png)

**Comment by jefferai** [comment]
Hi @Tristan971 ,

The issue was closed because the issues previous users had were either due to a configuration error or were sorted out. Please open a new issue. In there if you could provide more complete examples of the token and userinfo output from the account it would be great. Feel free to redact values as needed, but the structure would help.
